src/appleseed-max-impl/appleseedrenderer/dialoglogtarget.cpp

```diff
@@ -0,0 +1,259 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Sergo Pogosyan, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "dialoglogtarget.h"
+
+// appleseed-max headers.
+#include "appleseedrenderer/resource.h"
+#include "main.h"
+#include "utilities.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/windows.h"    // include before 3ds Max headers
+#include "foundation/utility/string.h"
+
+// Boost headers.
+#include "boost/thread/locks.hpp"
+#include "boost/thread/mutex.hpp"
+
+// 3ds Max headers.
+#include <max.h>
+
+// Standard headers.
+#include <Richedit.h>
+
+namespace asf = foundation;
+namespace asr = renderer;
+
+namespace
+{
+    boost::mutex                g_message_queue_mutex;
+    std::vector<MessagePair>    g_message_queue;
+    HWND                        g_log_dialog = 0;
+
+    static INT_PTR CALLBACK dialog_proc(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam)
+    {
+        switch (msg)
+        {
+          case WM_INITDIALOG:
+            break;
+          case WM_CLOSE:
+          case WM_DESTROY:
+            if (g_log_dialog != 0)
+            {
+                GetCOREInterface14()->UnRegisterModelessRenderWindow(g_log_dialog);
+                g_log_dialog = 0;
+                DestroyWindow(hwnd);
+            }
+            break;
+          
+          case WM_SIZE:
+          {
```

Comment:
Ident `{` like any other code under `case` (2 spaces). Then the contents of the `{}` block is indented like any other code (4 spaces).

    case WM_SIZE:
      {
          HWND edit_box = GetDlgItem(hwnd, IDC_EDIT_LOG);
          ...
      }
      break;